src/js/components/settings/webhooks/configuration.js

```diff
@@ -0,0 +1,103 @@
+  const {
+    [EXTERNAL_PROVIDER.webhook.credentialsType]: { secret: hookSecret = '', url: hookUrl = '' }
+  } = credentials;
+  useEffect(() => {
+    setDescription(hookDescription);
+    setUrl(hookUrl);
+    setSecret(hookSecret);
+  }, [adding, editing, hookDescription, hookSecret, hookUrl]);
+
+  useEffect(() => {
+    setSecretFormatError(!secret || validator.isHexadecimal(secret) ? '' : 'The secret has to be entered as a hexadecimal string');
+  }, [secret]);
+
+  const onSubmitClick = () => {
+    onSubmit({
+      id,
+      provider: EXTERNAL_PROVIDER.webhook.provider,
+      credentials: { type: EXTERNAL_PROVIDER.webhook.credentialsType, [EXTERNAL_PROVIDER.webhook.credentialsType]: { secret, url } },
+      description
+    });
+  };
+
+  const secretInputTip = editing ? 'Cannot edit webhook secrets after they have been saved' : 'The secret has to be entered as a hexadecimal string';
+  const descriptionInput = (
+    <TextField
+      label="Description (optional)"
+      id="webhook-description"
+      disabled={editing}
+      multiline
+      onChange={e => setDescription(e.target.value)}
+      value={description}
+    />
+  );
+  const isSubmitDisabled = !url || (editing && url === hookUrl);
+  return (
+    <>
+      <InfoHint content="Webhooks are triggered when a device's status is updated, or a device is decommissioned or provisioned." />
+      <div className="flexbox column" style={{ width: 500 }}>
+        <TextField label="Url" id="webhook-name" value={url} onChange={e => setUrl(e.target.value)} />
+        {editing ? (
+          <MenderTooltip arrow placement="bottom-start" title="Cannot edit webhook description after it has been saved">
+            {descriptionInput}
+          </MenderTooltip>
+        ) : (
+          descriptionInput
+        )}
+        <MenderTooltip arrow placement="bottom-start" title={secretInputTip}>
+          <TextField
+            label="Secret (optional)"
+            id="webhook-secret"
+            disabled={editing}
+            error={!editing && !!secretFormatError}
+            helperText={!editing && secretFormatError}
+            onChange={e => setSecret(e.target.value)}
+            type={editing ? 'password' : 'text'}
+            value={secret}
+          />
+        </MenderTooltip>
+      </div>
+      <Divider className="margin-top-large" light />
+      <div className="flexbox centered margin-top" style={{ justifyContent: 'flex-end' }}>
+        <Button onClick={onCancel}>Cancel</Button>
+        <Button color="secondary" variant="contained" disabled={isSubmitDisabled} onClick={onSubmitClick}>
+          {editing ? 'Save' : 'Submit'}
+        </Button>
+      </div>
+    </>
+  );
+};
+
+export const WebhookCreation = ({ adding, onCancel, ...props }) => (
+  <Drawer anchor="right" open={adding} PaperProps={{ style: { minWidth: 600, width: '50vw' } }}>
+    <div className="flexbox center-aligned margin-bottom-small space-between">
+      <h3>Add a webhook</h3>
```

Comment:
Q: Why not use `Typography`?

Reply:
while the Typography allows to more centrally manage the theming of anything typography I feel it dilutes the connection to the markup, since you can write `<Typography variant="h2" />` but define a different mapping to the proper html `<h2 />` in the theme… so you might end up writing `h2` but getting `h6` or `span`… if you know the entire codebase you would know, but if multiple people work on theme + components… the connection gets fuzzy too easily… 🤷‍♂️ not an ideal position for me to have I think…